src/components/dialogs/commons/use-name-check.tsx

```diff
@@ -0,0 +1,100 @@
+import CheckIcon from '@mui/icons-material/Check';
+import { useSelector } from 'react-redux';
+import { useDebounce } from '@gridsuite/commons-ui';
+import { FieldError } from 'react-hook-form';
+import { ElementType } from '../../../utils/elementType';
+import { ReduxState } from '../../../redux/reducer.type';
+
+export type NameCheckReturn = [ReactElement | null, boolean];
+
+interface INameCheckProps<T> {
+    field: keyof T;
+    value: string;
+    elementType: ElementType;
+    setError: (field: keyof T, error: FieldError) => void;
+}
+
+export const useNameCheck = <T extends Record<string, any>>({
+    field,
+    value,
+    elementType,
+    setError,
+}: INameCheckProps<T>): NameCheckReturn => {
+    const intl = useIntl();
+
+    const [adornment, setAdornment] = useState<ReactElement | null>(null);
+    const [isChecking, setIsChecking] = useState(false);
+
+    const activeDirectory = useSelector(
+        (state: ReduxState) => state.activeDirectory
+    );
+
+    const handleCheckName = useCallback(() => {
+        const valueFormatted = value.replace(/ /g, '');
+
+        if (!valueFormatted) {
+            setAdornment(null);
+        } else {
+            setAdornment(
+                <InputAdornment position="end">
+                    <CircularProgress size="1rem" />
+                </InputAdornment>
+            );
+
+            elementExists(activeDirectory, value, elementType)
+                .then((isElementExists) => {
+                    if (isElementExists) {
+                        setError(field, {
+                            type: 'nameAlreadyUsed',
+                            message: intl.formatMessage({
+                                id: 'nameAlreadyUsed',
+                            }),
+                        });
+                        setAdornment(null);
+                    } else {
+                        setAdornment(
+                            <InputAdornment position="end">
+                                <CheckIcon sx={{ color: 'green' }} />
+                            </InputAdornment>
+                        );
+                    }
+                })
+                .catch((error) =>
+                    setError(field, {
+                        type: 'nameValidityCheckErrorMsg',
+                        message:
+                            intl.formatMessage({
+                                id: 'nameValidityCheckErrorMsg',
+                            }) + (error as Error).message,
+                    })
+                );
+        }
+
+        setIsChecking(false);
+    }, [value, field, setError, intl, activeDirectory, elementType]);
+
+    const debouncedHandleCheckName = useDebounce(handleCheckName, 700);
+
+    // handle check case name
+    useEffect(() => {
+        setIsChecking(true);
+        debouncedHandleCheckName();
+
+        if (!value) {
+            setAdornment(null);
+        }
+    }, [debouncedHandleCheckName, value]);
```

Comment:
we have to make 
const debouncedHandleCheckName = (value) =>
        useDebounce(handleCheckName, 700);
        
        that is not possible because useDebounce is a hook and we can't put it in a function